tests/unit/replybufsize.tcl

```diff
@@ -0,0 +1,56 @@
+proc get_reply_buffer_size {cname} {
+
+    set clients [split [string trim [r client list]] "\r\n"]
+    set c [lsearch -inline $clients *name=$cname*]
+    if {![regexp rbs=(\[a-zA-Z0-9-\]+) $c - rbufsize]} {
+        error "field rbus not found in $c"
+    }
+    return $rbufsize
+}
+
+start_server {tags {"replybufsize"}} {
+    
+    test {idle client buffer maximum shrink to 1kib} {
+        variable rbs 0
+        
+        # Create a simple idle test client
+        variable tc [redis_client]
+        $tc client setname test_client
+        
+        # make sure the client is idle for 5 seconds to make it shrink the reply buffer
+        after 5000 {set rbs [get_reply_buffer_size test_client]} 
+        vwait rbs
+        
+        assert_equal $rbs 1024
+        
+        $tc close
+    }
+    
+    test {busy client buffer maximum enlarge to 16kib} {
+        variable rbs 0
+       
+        r set bigval [string repeat x 16384]
```

Comment:
@yoav-steinberg I initially started to check intermediate values. but I wanted to first introduce the basic cases.
the main problem is that the malloc tests will fail since it will not be aligned to the same size classes IMO.

Reply:
> the main problem is that the malloc tests will fail since it will not be aligned to the same size classes IMO.

I think just checking that the buffer's under 16k but over 1k is enough to verify "intermediates". Doesn't need to be exact or assume specific size classes.